src/utils/get-import-flavor-of-node.ts

```diff
@@ -0,0 +1,29 @@
+import {
+    importFlavorIgnore,
+    importFlavorRegular,
+    importFlavorSideEffect,
+    importFlavorType,
+} from '../constants';
+import type { GetImportFlavorOfNode } from '../types';
+
+/**
+ * Classifies nodes by import-flavor, primarily informing whether the node is a candidate for merging
+ *
+ * @param node
+ * @returns {("prettier-ignore"|"regular"|"side-effect"|"type")}
+ */
+export const getImportFlavorOfNode: GetImportFlavorOfNode = (node) => {
+    const hasIgnoreNextNode = (node.leadingComments ?? []).some(
+        (comment) => comment.value.trim() === 'prettier-ignore',
```

Comment:
Question, will this count as an ignored node?

```js
// prettier-ignore
// something else
import foo from 'foo';
```

Should we only consider the comment directly above the import?  Honest question, I'm not sure what the right answer is.

Reply:
I copied this helper method from `getChunkTypeOfNode` which this method was heavily inspired by.

I assumed that that was the "intentional" logic here, but if that logic is wrong. I'm totally fine hoisting both of them somewhere and changing it somehow.

Ref: https://github.com/IanVS/prettier-plugin-sort-imports/blob/3b72a60b254d8fab9e8c362b31be54d028309e30/src/utils/get-chunk-type-of-node.ts#L23-L25

Reply:
(I've now hoisted this out for sharing between the two locations)